libs-scala/scala-utils/src/main/scala/com/daml/scalautil/future/FutureConversion.scala

```diff
@@ -0,0 +1,29 @@
+package com.daml.scalautil.future
+
+import java.util.concurrent.{CompletionException, CompletionStage}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.jdk.javaapi.FutureConverters
+
+object FutureConversion {
+
+  /** Converts a java [[CompletionStage]] into a scala [[Future]]
+    * Java [[CompletionStage]] wraps any exception that causes failure into a [[CompletionException]]
+    * Compared to the standard library conversions found in [[scala.jdk.FutureConverters]]
+    * this conversion unwraps any failure and fails the future with the root cause found in the [[CompletionException]]
+    */
+  def toScalaUnwrapped[T](cs: CompletionStage[T]): Future[T] = {
+    FutureConverters
+      .asScala(cs)
+      .recover {
+        case ex: CompletionException if ex.getCause != null =>
+          throw ex.getCause
```

Comment:
Nit: I think I'm a bit allergic to `throw`. Isn't `recoverWith` + `Future.failed` more clean?

Reply:
I would disagree with that. If I see a `recoverWith` instead of `recover` I would expect that we have a meaningful async computation inside, and not using a `Future` to cover an exception.

Reply:
You could use `transform(identity, { case ex: CompletionException ... })` to make it clear you're simply transforming the failure.

Reply:
If I get an agreement that it makes it cleaner I'm all up for it.
For me, it still doesn't feel that much cleaner because of the following reasons:
1. It's still not clear that you transform only the failure (in some cases you could argue that it's quite the opposite, as you know that recover always handles only failures while transform can handle both success and failure) until you keep reading and see the identity call 
2. `transform` doesn't work with partial functions, therefore we would need to also handle the other cases...this can be done fairly easy by lifting the partial function which handles our transformations, but it's once again extra clutter

Reply:
Fine to me to leave it